packages/plugin-commands-installation/src/recursive.ts

```diff
@@ -263,6 +268,9 @@ export default async function recursive (
 
         const { manifest, writeProjectManifest } = manifestsByPath[rootDir]
         let currentInput = [...input]
+        if (cmdFullName === 'update' && currentInput) {
+          currentInput = getScopedInput(input, manifest, include)
+        }
```

Comment:
It will be more optimal if you create a matcher outside of this loop. Like 

```ts
const updateMatch = cmdFullName === 'update' && currentInput ? matcher(currentInput) : null
```

then in the loop you could just use it

```ts
if (updateMatch) {
  currentInput = matchDependencies(updateMatch, input, manifest, include)
}
```

From the performance perspective it will be good because the matcher already creates optimal matching functions. See https://github.com/pnpm/pnpm/blob/df21cf309cc654857f861243d85633406ac88499/packages/matcher/src/index.ts#L16-L18